python/health-check/src/health_check/exporters/supportconfig_exporter.py

```diff
@@ -0,0 +1,583 @@
+# SPDX-FileCopyrightText: 2023 SUSE LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+Main supportconfig exporter module that collects metrics into a json file
+and serves the file by using an HTTP server.
+"""
+
+from collections import defaultdict, namedtuple
+import os
+from pathlib import Path
+import re
+import signal
+import sys
+from typing import Dict, Tuple
+import yaml
+import json
+
+import http.server
+
+import static_metrics
+from static_metrics import metrics_config
+
+
+def sigterm_handler(**kwargs):
+    del kwargs  # unused
+    print("Detected SIGTERM. Exiting.")
+    sys.exit(0)
+
+
+signal.signal(signal.SIGTERM, sigterm_handler)
+
+
+class SupportConfigMetricsCollector:
+    """A collector that collects metrics, and exports them into a file."""
+
+    def __init__(self, supportconfig_path=None):
+        if not supportconfig_path:
+            raise ValueError("A 'supportconfig_path' must be set via config.yml file")
+
+        self.supportconfig_path = supportconfig_path
+        self.static_metrics_collection = (
+            static_metrics.create_static_metrics_collection(self.supportconfig_path)
+        )
+        self.disk_layout = []
+        self.salt_keys = []
+        self.salt_jobs = []
+        self.salt_configuration = []
+        self.max_clients = -1
+        self.server_limit = -1
+        self.num_of_channels = -1
+        self.shared_buffers_to_mem_ratio = -1
+        self.major_version = -1
+        self.fs_mount_insufficient = -1
+        self.fs_mount_out_of_space = -1
+        self.roles = [
+            {
+                "name": "master",
+                "value": 0,
+            },
+            {
+                "name": "proxy",
+                "value": 0,
+            },
+            {
+                "name": "client",
+                "value": 0,
+            },
+        ]
+        self.parse()
+
+    def parse(self):
+        if self.exists_salt_jobs_file():
+            self.salt_jobs = self.read_salt_jobs()
+        if self.exists_salt_keys_file():
+            self.salt_keys = self.read_salt_keys()
+        if self.exists_salt_configuration_file():
+            self.salt_configuration = self.read_salt_configuration()
```

Comment:
non-blocking, but these checks should be included in the methods they are guarding, e.g. `read_salt_jobs` should have a guart statement `if not self.exists_salt_jobs_file():` etc.

Reply:
Fixed now